src/main/java/org/sasanlabs/service/vulnerability/csrf/CSRFVulnerability.java

```diff
@@ -0,0 +1,218 @@
+package org.sasanlabs.service.vulnerability.csrf;
+
+import java.io.IOException;
+import java.util.Random;
+import java.util.function.Supplier;
+import org.apache.commons.lang3.StringUtils;
+import org.sasanlabs.internal.utility.LevelConstants;
+import org.sasanlabs.internal.utility.Variant;
+import org.sasanlabs.internal.utility.annotations.AttackVector;
+import org.sasanlabs.internal.utility.annotations.VulnerableAppRequestMapping;
+import org.sasanlabs.internal.utility.annotations.VulnerableAppRestController;
+import org.sasanlabs.service.vulnerability.bean.GenericVulnerabilityResponseBean;
+import org.sasanlabs.vulnerability.types.VulnerabilityType;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+/**
+ * This class contains vulnerabilities related to CSRF.
+ *
+ * @author xxx
+ */
+@VulnerableAppRestController(descriptionLabel = "CSRF_VULNERABILITY", value = "CSRFVulnerability")
+public class CSRFVulnerability {
+
+    private static final String RTN_BADINPUT = "Bad input!";
+
+    private static final String USR_LOGIN = "id";
+    private static final String USR_PASS = "pass";
+    private static final String USR_NEWPASS = "newpass";
+
+    private static final String LOGIN = "mylogin";
+    private String passlive = "mypassword";
+    private Boolean connect = false;
+
+    private static final String RTK_PARAM = "rtkparam";
+    private static final String FIXED_TOKEN = "7877864849";
+    private String scrt = "";
+
+    private static final Random rand = new Random();
+
+    private String checkInputs(
+            String usrId, String usrPass, String usrNewPass, Boolean securityOk) {
+        Supplier<Boolean> validatorId = () -> StringUtils.isNotBlank(usrId + usrPass);
+        Supplier<Boolean> validatorNP = () -> StringUtils.isNotBlank(usrNewPass);
```

Comment:
Suggestion, to make supplier as static and final such that it will not take heap memory again and again.